### ISSUE TITLE
Add pagination to facets on results pages

### ISSUE THREAD

**Issue by kamerynB** [project-lux/lux-frontend#252]
**Problem Description:**
Facets are currently causing performance issues. This issue will cover all facets. 

**Expected Behavior/Solution:**
ML will be returning the activity streams and pagination will now be required for facets containing lists.

5/29 UAT: For date dacets, FE will need two requests instead of one. One for the first and one for the last page. The results will be returned in order so that the first results on the first page is the earliest year of the range, and the last result on the last page will be the latest year of the range.

**Requirements:**

- [x] Each facet list will continue to have the "Show More" button
- [x] "Show More" will trigger a request to the backend for the results of the current page + 1

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [x] Wireframe/Mockup - Kam
- [x] ~Committee discussions - Sarah~
- [x] ~Feasibility/Team discussion - Sarah~
- [x] Backend requirements - Brent
- [x] Frontend requirements - Kam
- [x] ~Are new regression tests required for QA - Amy~
- [ ] Questions
    -  How will this impact date facets as they are not rendered as a list? A | 5/29 UAT; @clarkepeterf BE/MT ticket to see if the dates are returned in order.  New ticket will need to be made for date facets only.  

**UAT/LUX Examples:**

- Any search result
- TST: Andy Warhol Object Results | Object Type facet: https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22text%22%3A%22andy%22%2C%22_lang%22%3A%22en%22%7D%2C%7B%22text%22%3A%22warhol%22%2C%22_lang%22%3A%22en%22%7D%5D%7D&sq=andy+warhol
- Facet with over 700K results (categorized as fossil): https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22OR%22%3A%5B%7B%22classification%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fconcept%2F05a41429-8a18-4911-854e-eae804b7d46f%22%7D%7D%2C%7B%22material%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fconcept%2F05a41429-8a18-4911-854e-eae804b7d46f%22%7D%7D%5D%7D&openSearch=false&ip=1&if=%7B%22AND%22%3A%5B%7B%22classification%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux-front-tst.collections.yale.edu%2Fdata%2Fconcept%2F05a41429-8a18-4911-854e-eae804b7d46f%22%7D%7D%5D%7D&facetRequest=true

**Dependencies/Blocks:**

- Blocked By: https://github.com/project-lux/lux-marklogic/issues/160
- Blocked By: https://github.com/project-lux/lux-marklogic/issues/161
- Blocked By: https://github.com/project-lux/lux-middletier/issues/65
- Milestone Block By: https://github.com/project-lux/lux-marklogic/issues/162 

**Related Github Issues:**
- Date facets only: https://github.com/project-lux/lux-frontend/issues/262

**Related links:**
N/A

**Wireframe/Mockup:**
Wireframe is not needed as it will look and behave as it did before this change it completed.

**Comment by roamye** [comment]
#Agenda

**Comment by kamerynB** [comment]
5/31 decision regarding date facets and HAL links that require dates:
For facets on the results page, the frontend will make a request to both the first page and last page of the date results to get the earliest and latest dates. From there, the frontend can generate the range input fields with a minimum and maximum year.

For timeline lists, the frontend can pagination through the results in the same manner it will paginate on the results facets.

This will still impact the timeline graph but as that is not implemented yet, we will reevaluate solutions when it is necessary.

cc: @jffcamp @prowns @roamye @clarkepeterf @azaroth42

**Comment by roamye** [comment]
From 5/31 IT meeting: 

If Performance Test | https://github.com/project-lux/lux-marklogic/issues/162, passes then we can add this tix to the 6/10 milestone. 

Separate ticket for the date facets ONLY is: https://github.com/project-lux/lux-frontend/issues/262

**Comment by roamye** [comment]
Approved by UAT

**Comment by roamye** [comment]
reviewed 7/16  Looked good in prod, closing.
